feat<CI>:Integration of Gitlab CI for build and tests.

yudaykiran opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you -->

**What this PR does / why we need it**:
- Added the gitlab yaml to track commits and build in gitlab
- Gitlab yaml triggers subsequent pipelines to run litmus tests.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:
Signed-off-by: Uday Kiran Y <uday.kiran@openebs.io>

Commit:
Integration of Gitlab CI for build and tests.

- Added the gitlab yaml to track commits and build in gitlab
- Gitlab yaml triggers subsequent pipelines to run litmus tests.

Signed-off-by: Uday Kiran Y <uday.kiran@openebs.io>

vishnuitta reviewed (approved):
compilation and image build looks good ..
@atulabhi can you please review baseline stage?

Commit:
Update the git configuration before pushing changes to baseline.

- The git configuration needs to be updated before making any pushes to
the baseline repo.

Signed-off-by: Uday Kiran Y <uday.kiran@openebs.io>

pawanpraka1 committed:
Merge branch 'replication' into AddGitlabIntegration

pawanpraka1 committed:
Merge branch 'replication' into AddGitlabIntegration
